lib/elixir/lib/code.ex

```diff
@@ -1197,6 +1197,105 @@ defmodule Code do
     :elixir.string_to_quoted!(to_charlist(string), line, column, file, opts)
   end
 
+  @doc """
+  Converts the given string to its quoted form and a list of comments.
+
+  This function is useful when performing textual changes to the source code,
+  while preserving information like comments and literals position.
+
+  Returns `{:ok, quoted_form, comments}` if it succeeds,
+  `{:error, {line, error, token}}` otherwise.
+
+  Comments are maps containing information about the line they were found, its
+  contents, and how many end of lines were found between the comment and the
+  closest tokens:
+      iex> Code.string_to_quoted_with_comments("\""
+      ...> :foo
+      ...>
+      ...> # Hello, world!
+      ...>
+      ...>
+      ...> # Some more comments!
+      ...> "\"")
+      {:ok, :foo, [
+        %{line: 3, previous_eol: 2, next_eol: 3, text: "\# Hello, world!"},
+        %{line: 6, previous_eol: 3, next_eol: 1, text: "\# Some more comments!"},
```

Comment:
Let's also add an example where previous_eol is zero. It can be a separate example so this one does not grow to large. Something like: `Code.string_to_quoted_with_comments(":foo # :bar")`.